numpy/core/src/multiarray/nditer_constr.c

```diff
@@ -692,6 +706,13 @@ NpyIter_Deallocate(NpyIter *iter)
         }
     }
 
+    if (iter->casterr_type != NULL) {
+        /* A casting error occurred, set it and do not resolve writeback */
```

Comment:
Should the iter object still be freed?

Reply:
Of course?  This is like a `Py_DECREF` call, with the problem, that I don't know what to do with the error which should be set already when this gets called, but for which we have no API to notify the user about that it happened.

Reply:
But the writeback-if-copy makes me a bit hesitant. Although, it may not matter whether or not it is performed (but it cannot be performed with an error set). I guess there is some potential for uninitialized stuff to be called back, but probably that is safe (you got an error after all, so should expect undefined behaviour).

Reply:
Maybe you should continue through the `PyArray_DiscardWritebackIfCopy` calls on the ndarrays via setting `resolve = 0`

Reply:
Ooops...